### ISSUE TITLE
Bug: Nested Collection of any type

### ISSUE THREAD

**Issue by mark-todd** [rutek/dataclass#6]
Slightly odd bug - when a collection of substructures with an any in it gets parsed through dataclass it produces a duplicate structure inside - easier to show the example than describe:

```php
class Substructure {
    public int $num;
    public $anytype;
}

use Rutek\Dataclass\Collection;

class SubstructureCollection extends Collection {
    public function __construct(Substructure ...$str)
    {
        $this->items = $str;
    }
}

class TestRoute {
    public SubstructureCollection $substructures;
}
```
Suppose then a json request body like this is supplied:
```json
{
  "substructures": [
    {
      "num": 1,
      "anytype": 1
    }
  ]
}
```
Once processed with `transform`, it becomes:
```json
{
   "substructures":[
      {
         "num":1,
         "anytype":{
            "num":1,
            "anytype":1
         }
      }
   ]
}
```

**Comment by rutek** [comment]
Thank you for submitting this issue. It has been fixed in v0.1.1 which has been just released. Have a nice day! :)

**Comment by mark-todd** [comment]
Thanks that's great! You too :)
